### PR TITLE
add 76

### PR DESCRIPTION
**Pull request by JoannaCode**
@FreeTymeKiyan

### CODE REVIEW

File: Hard/MinimumWindowSubstring.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ */
+
+public class MinimumWindowSubstring {
+    public static String minWindow(String s, String t) {
+        if(s.length() == 0 || t.length() ==0 || s.length() < t.length()) {
+            return "";
+        }
+        HashMap<Character, Integer> map = new HashMap<Character, Integer>();
+        for(char c : t.toCharArray()) {
+            if(map.containsKey(c)) {
```
Comment (FreeTymeKiyan) [inline Hard/MinimumWindowSubstring.java]: can be more concise like `map.put(c, (map.containsKey(c) ? map.get(c) : 0) + 1);`

File: Hard/MinimumWindowSubstring.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                        minStart = left;
+                        minLength = right - left + 1;
+                    }
+                    if(map.containsKey(s.charAt(left))) {
+                        map.put(s.charAt(left), map.get(s.charAt(left)) + 1 );
+                        if(map.get(s.charAt(left)) > 0) {
+                            count--;
+                        }
+                    }
+                    left ++;
```
Comment (FreeTymeKiyan) [inline Hard/MinimumWindowSubstring.java]: there is a space between "left" and "++", should be removed

**Comment by JoannaCode** [comment]
@FreeTymeKiyan  fixed!

**Comment by FreeTymeKiyan** [comment]
:+1:

**Pull request merged**